### PR TITLE
Force LGTM to test on Python instead of legacy Python

### PR DESCRIPTION
**Pull request by cclauss**
A subset of #3273

### CODE REVIEW

**Comment by cclauss** [comment]
Because a question was asked at https://github.com/curl/curl/pull/3273#issuecomment-439025528 that we want to be able to answer.  The main PR does two things: 1. Force LGTM to run on Python 3 and 2. Add flake8 tests to find syntax errors.  The question was basically "Can LGTM find syntax errors without flake8?" and this PR will help to answer that.

**Comment by bagder** [comment]
OK, I didn't see that question but sure. If it does cause errors to be detected we want them fixed within the same PR that detects them. We don't want to merge code that knowingly causes CI failures.

**Comment by infinnovation-dev** [comment]
Testing against python3 will most likely fail because the impacket
code uses python2 constructs.  I guess going forward we should aim to make code python3 compatible.  The upstream impacket project has [an open issue](https://github.com/SecureAuthCorp/impacket/issues/61) for porting but there doesn't seem to be much progress.

**Comment by bagder** [comment]
Thanks @infinnovation-dev for the added info.

This makes it pretty clear what needs to happen before we can merge a PR like this...

**Comment by sjvs** [comment]
I've just answered @cclauss's question about the JavaScript analysis that was kicked off for this PR: https://discuss.lgtm.com/t/why-does-adding-python-also-add-javascript/1544/2

tl;dr: when the `lgtm.yml` changes, we run all the analyses for all languages that were once detected in a project. At some point, curl indeed contained (very little) JavaScript, so the analysis for that language was attempted. It failed, because it couldn't find any JavaScript. Nothing to worry about.

On a related note: LGTM will automatically filter out alerts from test code; such results are often irrelevant because test code is intentionally written to attempt to break things. @cclauss's PR https://github.com/curl/curl/pull/3273 suggests that you're interested in these results. They are actually available on LGTM's "Files" tab when you click "show excluded files": https://lgtm.com/projects/g/curl/curl/latest/files/?sort=name&dir=ASC&mode=heatmap&showExcluded=true

If you'd like to include results for test code in the automated code review for pull requests, the best thing to do is to manually classify the `tests` directory as regular code: https://lgtm.com/help/lgtm/customizing-file-classification

**Comment by bagder** [comment]
I think we should keep hiding the tests from the python checker. I don't see any upside in us rescanning those files over and over.

Personally, I would disable the Python and Javascript checks completely if we could. They just add two useless lines to every commit and PR.

1. The Python code we have is imported and only used for testing.
2. We don't have javacript in our products.

**Comment by infinnovation-dev** [comment]
> LGTM will automatically filter out alerts from test code

Aha! That explains why LGTM didn't pick up the breakage to negtelnetserver.py caused by the ill-considered PR #3252 .

**Comment by sjvs** [comment]
@infinnovation-dev: indeed, as I think you pointed out yourself on that PR: https://github.com/curl/curl/pull/3252#issuecomment-437191616

**Comment by asolino** [comment]
> Testing against python3 will most likely fail because the impacket
> code uses python2 constructs. I guess going forward we should aim to make code python3 compatible. The upstream impacket project has [an open issue](https://github.com/SecureAuthCorp/impacket/issues/61) for porting but there doesn't seem to be much progress.

That is wrong. 
The Python3 port is moving along well. You can use [this branch](https://github.com/SecureAuthCorp/impacket/tree/python36). Testing is needed so help is appreciated.

**Comment by infinnovation-dev** [comment]
> That is wrong.
> The Python3 port is moving along well.

Sorry, my bad.  I've added a comment to SecureAuthCorp/impacket#61 to link the python3 issues of the two projects together.

**Comment by cclauss** [comment]
The problem is that the "big bang" approach to porting to Python 3 is not often successful for large codebases especially [without continuous integration testing](https://github.com/SecureAuthCorp/impacket/issues/482).  The python36 branch referred to above is _107 commits ahead, 4 commits behind master_ and the porting process has been going for at least two months.  There is great work on that branch there but it is difficult to review so many changes all at once.  A more incremental approach across multiple PRs would allow multiple contributors to participate in the porting while always ensuring that the changes do not break Python 2 compatibility.
